script.js

```diff
@@ -1,12 +1,121 @@
-//You can edit ALL of the code here
+// level 350
+let allEpisodes; // global scope to be accessible from inside all functions
+
+function fetchAllEpisodes() {
+  return fetch("https://api.tvmaze.com/shows/82/episodes")
+    .then((response) => response.json())
+    .then((data) => {
+      allEpisodes = data; // this was done to move allEpisodes to global scope
+      makePageForEpisodes(allEpisodes);
+    })
+    .catch(function (error) {
+      console.log(error);
+    });
+}
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  fetchAllEpisodes();
+  // makePageForEpisodes(allEpisodes);
 }
 
 function makePageForEpisodes(episodeList) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  //rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+  // level 100
+  document.getElementById("root").innerHTML = "";
+  for (i = 0; i < episodeList.length; i++) {
+    // step 1 - create elements
```

Comment:
i really like how you have used comments on each stage of your code. it makes it much easier to follow your code and its it going something i will try to implement on my project.